engine/plugins/fbs/fbs.js

```diff
@@ -285,90 +286,76 @@ module.exports = function(options, imports, register) {
-                        onlineState.online = true;
+                    if (result.onlineStatus === 'Y') {
+                        deferred.resolve(result);
                     } else {
-                        // FBS online but threshold _not_ reached, so state offline.
-                        onlineState.successfulOnlineChecks++;
-                        onlineState.online = false;
-                        checkOnlineStateTimeout = setTimeout(checkOnlineState, onlineState.offlineTimeout);
+                        deferred.reject(new Error('FBS is offline'));
                     }
-                } else {
-                    // FBS is offline, so it the state.
-                    onlineState.successfulOnlineChecks = 0;
-                    onlineState.online = false;
-                    checkOnlineStateTimeout = setTimeout(checkOnlineState, onlineState.offlineTimeout);
-                }
-
-                // Send state event into the bus.
-                const eventName = onlineState.online ? 'fbs.online' : 'fbs.offline';
-                bus.emit(eventName, {
-                    timestamp: new Date().getTime(),
-                    online: onlineState
-                });
-            },
-            () => {
-                // Error connecting to FBS.
-                onlineState.online = false;
-                onlineState.successfulOnlineChecks = 0;
-                checkOnlineStateTimeout = setTimeout(checkOnlineState, onlineState.offlineTimeout);
-                bus.emit('fbs.offline', {
-                    timestamp: new Date().getTime(),
-                    online: onlineState
+                })
+                .catch(error => {
+                    deferred.reject(error);
                 });
-            }
-        );
-    };
 
-    // Start the online checker.
-    if (enableOnlineChecks) {
-        checkOnlineState();
+            return deferred.promise;
+        };
+
+        const brake = new Brakes(checkFBSOnline, {
+            timeout: onlineState.requestTimeout,
+            threshold: onlineState.threshold,
+            circuitDuration: onlineState.circuitDuration,
+            waitThreshold: onlineState.waitThreshold
+        });
+
+        // Error state.
+        brake.on('circuitOpen', function() {
+            onlineState.online = false;
+            bus.emit('fbs.offline', {
+                timestamp: new Date().getTime(),
+                online: onlineState
+            });
+        });
+
+        // Setup online check interval.
+        setInterval(() => {
+            if (!brake.isOpen()) {
+                brake.exec('onlineCheck').then(
+                    () => {
+                        onlineState.online = true;
+                        bus.emit('fbs.online', {
+                            timestamp: new Date().getTime(),
+                            online: onlineState
+                        });
+                    },
+                    () => {}
```

Comment:
Why an empty error function? No need for it if you don't do anything with it.

Reply:
It is added here to avoid getting uncaught exceptions output to the log. Perhaps we should just add a log message here?

Reply:
I have added debug messages. Also to this case